solutions/gold/ioi-05-garden.mdx

```diff
@@ -0,0 +1,96 @@
+---
+id: ioi-05-garden
+source: IOI 2005
+title: 2005 - Garden
+author: Justin Ji
+---
+
+## Explanation
+
+Note that between any two disjoint rectangles, we can draw either a vertical
+or horizontal line between these two rectangles. We can consider every possible
+vertical and horizontal line, and find the two best rectangles between the
+lines.
+
+Let's first consider every vertical line. Consider fixing the two rows of our
+rectangle. If we can calculate the best rectangle perimeter for
+every prefix and suffix of our available columns, we can find our answer by
+sweeping over the splitting point. The implementation below iterates on the
+right column and then uses two pointers to find the best left column.
+Similar logic can be applied to find the best result for every horizontal line.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(l^{2}w + wl^{2})$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int l, w;
+	cin >> l >> w;
+	int n, k;
+	cin >> n >> k;
+	vector<vector<int>> grid(l + 1, vector<int>(w + 1));
+	for (int i = 0; i < n; i++) {
+		int x, y;
+		cin >> x >> y;
+		++grid[x][y];
+	}
+	const int INF = 1e9;
```

Comment:
yeah

Reply:
you can't use INT_MAX on this, otherwise it will overflow

Reply:
the real question is why you're doing operations on integers that could be INF

Reply:
```cpp
for (int i = 2; i <= w; i++) {
	res = min(res, pref_min[i - 1] + suff_min[i]);
}
```
this is the line in question where it can overflow. I personally think it's easier to have 1e9 than to manually check if its infinite.

Reply:
alright then add a comment
"we use 1e9 as INF to prevent overflow in this line"
also then just set res as int32_max

Reply:
done